Task 4, Node.js, Mongo, Passport, Socket.io

oddestdan opened the pull request:
Mongo integration TODO:

- [x] Use actual data from DB everywhere
- [ ] Add error handler
- [ ] Move strings to constants
- [ ] Establish WS connection/support
- [x] Cleanup module imports related code

oddestdan committed:
Task 4, Part 1 - Node.js, Socket.IO, Scheduler

oleksandr-zhynzher commented on packages/server/src/index.ts:
```diff
@@ -1,15 +1,32 @@
+
+const app = express();
+const server = http.createServer(app);
+const PORT = 8080;
```
Better to move such const to environment variables

oleksandr-zhynzher commented on packages/server/src/routes/api/achievement.ts:
```diff
@@ -0,0 +1,27 @@
+const router = express.Router();
+
+router.get('/achievement/:id', (req, res) => {
+  try {
```
Are we really need this try/catch here?

oddestdan committed:
Mongo integration and connection

1. Add Task, Achievement, Challenge and User
2. Prepare script for seeding initial DB data
3. Add `/login` endpoint with Passport and JWT
4. Add endpoint for current challenge for user
5. Secure all endpoints under JWT Bearer Token

TODO:

[ ] Use actual data from DB everywhere
[ ] Add error handler
[ ] Move strings to constants
[ ] Establish WS connection/support
[ ] Cleanup module imports related code

oddestdan committed:
Apply DB data for all requests, interfaces cleanup

oleksandr-zhynzher commented on packages/server/src/lib/auth.ts:
```diff
@@ -0,0 +1,46 @@
+    { usernameField: 'email', passwordField: 'password' },
+    async (email, password, done) => {
+      try {
+        const user = await UserModel.findOne({ email, password });
```
it`s better to split it in two steps. 1 find user by Email, 2. Validate password

oleksandr-zhynzher commented on packages/server/src/routes/api/task.ts:
```diff
@@ -0,0 +1,43 @@
+    const taskArchive = getTaskArchive(req.params.id, challenges);
+
+    if (!taskArchive) {
+      return res.json({ status: `Unable to find task archive` });
```
better to add status code for example 404
